File: poo/poligono/Ponto.cpp
#include <iostream>
#include "Ponto.h"

using namespace std;

Ponto::Ponto(float x, float y){}

Ponto::Ponto(){
  cout << "\n\tPonto criado!...\n";
}

Ponto::~Ponto()
{
  cout << "\n\tPonto destruido!...\n";
}

void Ponto::lerPonto()
{
  cout << "\n\tDigite as coordenadas do pornto: ";
  cout << "\nX: ";
  cin >> x;
  cout << "\n\tY: ";
  cin >> y;
}

string Ponto::escrevePonto()
{
  return to_string(x) + ", " + to_string(y);
}

float Ponto::distancia(Ponto p)
{
  return sqrt(pow(x - p.getX(), 2) + pow(y - p.getY(), 2));
}

bool Ponto::operator==(Ponto p)
{
  return p.getX() == x && p.getY() == y;
}

Ponto Ponto::operator+(Ponto p)
{
  Ponto p1(p.getX() + x, p.getY() + y);

  return p1;
}

File: projetos/src/semana5/atvEmSala/poligono/Ponto.java
package semana5.atvEmSala.poligono;

public class Ponto {
    
    private float x;
    private float y;

    public Ponto() {
    }

    public Ponto(float x, float y) {
        this.x = x;
        this.y = y;
    }

    public float getX() {
        return this.x;
    }
    public void setX(float x) {
        this.x = x;
    }
    public float getY() {
        return this.y;
    }
    public void setY(float y) {
        this.y = y;
    }

    @Override
    public String toString() {
        return "{" +
            " x='" + getX() + "'" +
            ", y='" + getY() + "'" +
            "}";
    }
}
